### PR TITLE
Fix Headless --import --quit crashes editor

### LINKED ISSUE #98062: Headless --import --quit crashes editor
### Tested versions

Reproduceable in 4.4.dev [842f98239713fd10cfd648cd6aa3781895f289eb]

### System information

Linux Pop OS 22.04

### Issue description

When importing with --headless --quit args, the godot editor will crash.  This appears to be due to unload_editor_addons() method that was introduced to being called in ``Main::iteration`` here:

https://github.com/godotengine/godot/commit/c151c7dce0b3802ddf2caa2bb2d6a227d7326c3f

The solution is simple.  Simply clear the ``addon_name_to_plugin`` so that we don't try to memdelete things that have already had memdelete called on them.  There is no need to keep track of these plugins in a list after they have been deleted from memory (at least so far as I can tell).  I did this in my fork over here and will be putting in a PR for it: https://github.com/Lange-Studios/godot/commit/96ab05e89ff7d4f9f2865b04d3b7bf2b489b009f

### Steps to reproduce

I suspect this may be a more challenging one to reproduce.  Likely it will not occur if the editor exits after the first time ``unload_editor_addons`` is called.  I suspect this to be a race condition that I somehow introduced in my project.  But if you can get it into this state, you should get segfault crashes due to ``remove_editor_plugin`` being called on plugins that have already had ``memdelete`` called.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by TCROC**
Fixes this issue here:

https://github.com/godotengine/godot/issues/98062

``addon_name_to_plugin.clear();`` should be called due to the fact each plugin in ``addon_name_to_plugin`` was just deleted in the preceding for loop.  Future attempts to access the items in ``addon_name_to_plugin`` after ``memdelete`` was called result in crashes.

* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/98062*

### CODE REVIEW

**Review by jamie-pate — APPROVED** [review]

**Review by clayjohn — APPROVED** [review]
makes sense to me. Would be good to have @bruvzg's approval too before merging

**Comment by bruvzg** [comment]
> If Main::iteration() has deleted all the plugns then it seems like import tasks relying on those plugins will either

It is deleting plugins only when exiting, and `Main::iteration` have a check to wait for running importers and unset `exit` flag, this should not be possible.

**Review by bruvzg — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**
